### PR TITLE
Exposed tooltip longPress action when available

### PR DESCRIPTION
**Pull request by harperl-lgtm**
This PR exposed tooltip longPress action when its trigger mode is longPress

Issues fixed:
[b/239606811]

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/tooltip.dart [hunk 1]
```diff
@@ -747,13 +748,24 @@ class TooltipState extends State<Tooltip> with SingleTickerProviderStateMixin {
...
+          child: result,
+        );
+      }
+      else {
+        // if _triggerMode is not tap exclude GestureDetector from semantics
+        // otherwise in cases the tooltip is used with a button the tap action
+        // on the button will be lost.
+        result = GestureDetector(
+          behavior: HitTestBehavior.opaque,
+          onLongPress: (_triggerMode == TooltipTriggerMode.longPress) ? _handlePress : null,
```
Comment (chunhtai) [inline packages/flutter/lib/src/material/tooltip.dart:764]: why create a GestureDetectorif the  _triggerMode != TooltipTriggerMode.longPress?

It looks like using a switch(_triggerMode) and handle each case separately will be cleaner?

File: packages/flutter/lib/src/material/tooltip.dart [hunk 2]
```diff
@@ -747,13 +748,24 @@ class TooltipState extends State<Tooltip> with SingleTickerProviderStateMixin {
...
-      result = GestureDetector(
-        behavior: HitTestBehavior.opaque,
-        onLongPress: (_triggerMode == TooltipTriggerMode.longPress) ? _handlePress : null,
-        onTap: (_triggerMode == TooltipTriggerMode.tap) ? _handleTap : null,
-        excludeFromSemantics: true,
-        child: result,
-      );
+      if (_triggerMode == TooltipTriggerMode.tap) {
+        result = GestureDetector(
+          behavior: HitTestBehavior.opaque,
```
Comment (chunhtai) [inline packages/flutter/lib/src/material/tooltip.dart:753]: should this has `excludeFromSemantics: _excludeFromSemantics` ?

File: packages/flutter/lib/src/material/tooltip.dart [hunk 3]
```diff
@@ -747,13 +748,24 @@ class TooltipState extends State<Tooltip> with SingleTickerProviderStateMixin {
...
+          onTap: _handleTap,
+          child: result,
+        );
+      }
+      else {
+        // if _triggerMode is not tap exclude GestureDetector from semantics
+        // otherwise in cases the tooltip is used with a button the tap action
+        // on the button will be lost.
```
Comment (chunhtai) [inline packages/flutter/lib/src/material/tooltip.dart:761]: I am not sure if I understand the reasoning. If this does not exclude semantics, we won't need to create a semantics in line 741 right?

**Review by chunhtai — COMMENTED** [review]
Several questions.
1. Why does this only apply to longpress? If the trigger mode is tap and it should still be announce double tap to activate, no?
2. It looks like this pr only solve a half a the problem described in the internal bug. 
> When the tooltip is triggered, the tooltip information must be read to talkback users.

Although I am not sure if this is valid, maybe need to check with native android behavior. We were explicitly told not to use announcement API to announce message, and only use the https://developer.android.com/reference/android/view/accessibility/AccessibilityNodeInfo#setTooltipText(java.lang.CharSequence)

**Comment by harperl-lgtm** [comment]
> Several questions.
> 
> 1. Why does this only apply to longpress? If the trigger mode is tap and it should still be announce double tap to activate, no?
> 2. It looks like this pr only solve a half a the problem described in the internal bug.
> 
> > When the tooltip is triggered, the tooltip information must be read to talkback users.
> 
> Although I am not sure if this is valid, maybe need to check with native android behavior

Thank you so much for the comments!!!
1. Yes I totally agree if the triggerMode is tap double tap to activate should always be announced, but what I'm trying to clarify here is that in cases where the tooltip is used with a button we also want this announcement even if the triggerMode is not tap, which will not be the case if the GestureDetector is not excluded from semantics.
2. Native Android does not announce tooltip message when tooltip is triggered. Thus keeping consistent with them here. Will comment on the bug about this.

**Pull request closed without merging**